lib/record_store/provider/ns1.rb

```diff
@@ -219,6 +227,23 @@ def symbolize_keys(hash)
       def secrets
         super.fetch('ns1')
       end
+
+      def api_rate_limit
+        uri = URI('https://api.nsone.net/v1/monitoring/regions')
+        req = Net::HTTP::Get.new(uri)
+        req['X-NSONE-Key'] = secrets['api_key']
+        res = Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
+          http.request(req)
+        end
+
+        ratelimit_remaining = res.to_hash["x-ratelimit-remaining"].first.to_i
+        ratelimit_period = res.to_hash["x-ratelimit-period"].first.to_i
+
+        if ratelimit_remaining < 2
+          sleep(ratelimit_period)
+        end
+        sleep(ratelimit_period / ratelimit_remaining)
```

Comment:
🤔 on line 242, if we have a low number of tokens available, we'll sleep for the entire ratelimit_period (fair).  Do you also intend to sleep for possibly the entire rate limit?   Feels like lines 243 and 245 shouldn't both run, but lemme know what ur intentions are.

Reply:
Good catch! In [`ns1-go`](https://github.com/ns1/ns1-go/blob/da571caa8ebc28580715f677f97799aefde48679/rest/client.go#L290-L295) they use a separate method that "returns" to ensure we're only doing one of the cases. Here, a `else` would probably be sufficient.

Reply:
Right, that's right. I mixed a bit with Go. As Raphaël mentioned, `else` should be sufficient.